shared/src/main/scala/scodec/codecs/ZlibCodec.scala

```diff
@@ -13,9 +15,38 @@ private[scodec] class ZlibCodec[A](codec: Codec[A], level: Int, strategy: Int, n
     }
 
   def decode(b: BitVector) =
-    b.bytes.inflate(chunkSize).fold(
+    inflate(b, chunkSize).fold(
       e => Attempt.failure(Err(e.getMessage)),
-      bb => codec.decode(bb.bits)
+      bb => codec.decode(bb.value.bits).map(_.mapRemainder(_ ++ bb.remainder))
```

Comment:
I agree, can't imagine a situation where it would be useful concatenated like that, it should be dropped. It's still possible to access the remainder by using `~ bits` codec or something.